utils.py

```diff
@@ -453,3 +454,30 @@ def get_all_language_codes_and_names():
 def unescape_encoded_uri_component(escaped_string):
     """Unescape a string that is encoded with encodeURIComponent."""
     return urllib.unquote(escaped_string).decode('utf-8')
+
+
+CACHE_SLUG = None
+def get_cache_slug():
+    """Returns appropriate cache slug depending whether dev or prod. This cache
+    slug should only be used for urls in html, css and js files since it
+    contains a forward slash.
+    """
+    global CACHE_SLUG # pylint: disable=global-statement
+    if CACHE_SLUG is None:
+        # pylint: disable=redefined-outer-name
+        if feconf.DEV_MODE and not feconf.IS_MINIFIED:
+            if feconf.CACHE_SLUG_DEV:
+                CACHE_SLUG = '/%s' % feconf.CACHE_SLUG_DEV
+            else:
+                CACHE_SLUG = ''
+        else:
+            yaml_file_content = dict_from_yaml(
+                get_file_contents('cache_slug.yaml'))
+            CACHE_SLUG = '/build/%s' % yaml_file_content['cache_slug']
+
+    return CACHE_SLUG
+
+
+def get_unique_id():
+    """Returns a unique id."""
+    return str(random.randint(1000, 9999))
```

Comment:
OK, well :) 6 or 8 chars, alphanumeric? Then you get 36^6 or 36^8 which should be large enough.